hadoop-tools/hadoop-aws/src/test/java/org/apache/hadoop/fs/s3a/auth/delegation/TestS3ADelegationTokenSupport.java

```diff
@@ -76,7 +76,7 @@ public void testSessionTokenDecode() throws Throwable {
         renewer,
         new URI("s3a://anything/"),
         new MarshalledCredentials("a", "b", ""),
-        new EncryptionSecrets(S3AEncryptionMethods.SSE_S3, ""),
+        new EncryptionSecrets(S3AEncryptionMethods.SSE_S3, "", ""),
```

Comment:
extend test to verify the secrets are round tripped correctly

Reply:
Not sure if this is was you meant, but in the next revision I will add values to the key and encryption context instead of empty string. Then I will assert that they match the valyes in the decoded identifier.

I will do the same in `testSessionTokenIdentifierRoundTrip()`